src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs

```diff
@@ -207,63 +207,127 @@ public Task<Either<ActionResult, DeleteReleasePlanViewModel>> GetDeleteReleaseVe
                 .OnSuccessDo(async () => await _processorClient.BulkDeleteDataSetVersions(releaseVersionId))
-                .OnSuccessDo(async release => await _cacheService.DeleteCacheFolderAsync(
-                    new PrivateReleaseContentFolderCacheKey(release.Id)))
+                .OnSuccessDo(async release => await _cacheService.DeleteCacheFolderAsync(new PrivateReleaseContentFolderCacheKey(release.Id)))
                 .OnSuccessDo(async () => await _releaseDataFileService.DeleteAll(releaseVersionId))
                 .OnSuccessDo(async () => await _releaseFileService.DeleteAll(releaseVersionId))
                 .OnSuccessVoid(async releaseVersion =>
                 {
-                    // NOTE: As only removing unapproved amendments, releaseVersion.Release will not be orphaned
-                    // and associated item in releaseVersion.Publication.ReleaseSeries will also not be orphaned
-                    // so no need to remove them.
-
-                    releaseVersion.SoftDeleted = true;
-                    _context.ReleaseVersions.Update(releaseVersion);
-
-                    var roles = await _context
-                        .UserReleaseRoles
-                        .AsQueryable()
-                        .Where(r => r.ReleaseVersionId == releaseVersionId)
-                        .ToListAsync();
-                    roles.ForEach(r => r.SoftDeleted = true);
-                    _context.UpdateRange(roles);
-
-                    var invites = await _context
-                        .UserReleaseInvites
-                        .AsQueryable()
-                        .Where(r => r.ReleaseVersionId == releaseVersionId)
-                        .ToListAsync();
-                    invites.ForEach(r => r.SoftDeleted = true);
-                    _context.UpdateRange(invites);
-
-                    var methodologiesScheduledWithRelease =
-                        GetMethodologiesScheduledWithRelease(releaseVersionId);
-
-                    // TODO EES-2747 - this should be looked at to see how best to reuse similar "set to draft" logic
-                    // in MethodologyApprovalService.
-                    methodologiesScheduledWithRelease.ForEach(m =>
+                    if (!releaseVersion.Amendment && releaseVersion.ApprovalStatus == ReleaseApprovalStatus.Draft)
                     {
-                        m.PublishingStrategy = Immediately;
-                        m.Status = Draft;
-                        m.ScheduledWithReleaseVersion = null;
-                        m.ScheduledWithReleaseVersionId = null;
-                        m.Updated = DateTime.UtcNow;
-                    });
+                        await HardDeleteForDraft(releaseVersion, cancellationToken);
+                    }
+                    else
+                    {
+                        await SoftDeleteForAmendment(releaseVersion, cancellationToken);
+                    }
```

Comment:
I kept them separate as I didn't want to change the existing functionality at this stage. [EES-1295](https://dfedigital.atlassian.net/browse/EES-1295) defines a task to properly delete everything.

Reply:
I take your point about handling orphaned files - if we stopped soft deleting releases and blob files aren't being cleaned up, those are going to be harder to clean up.

If we're going to leave this for now, I'd add a comment here referring to EES-1295 and explaining that the only reason we're not hard deleting everything is because we think we might be orphaning blob storage files and we want to sort that out before switching to just hard delete for everything.

I think you're strictly speaking right that the scope of this ticket is about removing draft releases, not sorting out the soft delete/hard delete stuff - but I'd still be inclined to sort it because it's relatively easy and it leaves the code in a good state. My experience of situations like this is that tickets like EES-1295 will not get picked up anytime soon - and maybe even never.

I'm going to ask for a third opinion, because part of me thinks it is easy enough that we should just do it now and the potential orphaning of stuff isn't a big deal. I'd be interested in what others think.

Reply:
Hey there.  I think you've both got valid points here, but I would probably err on the side of Tom here just because:

1) It doesn't allow the scope of the ticket to creep.
2) It would remove the only functionality that makes use of the soft delete flag, and leave a bunch of soft deleted information in the database and the model, with no trace back as to why it was there in the first place.

I'd be tempted instead to raise a ticket to request we get the time to remove this functionality altogether, and marry it up with a migration to remove soft deleted Releases from the codebase and the columns from the model.  I get what you're saying Mark about the fact the you wouldn't be confident that this would be prioritised - you could be right for sure - but we could frame it in light of the fact that it's adding complexity to this part of the codebase and also building up junk data that no-one wants!  That'd be my personal preference, and adding a ticket ref to the codebase here to say that we're wanting to remove the soft delete in EES-XXXX.